common/error/error.rs

```diff
@@ -160,51 +141,81 @@ macro_rules! typedb_error {
 
             fn code_number(&self) -> usize {
                 match self {
-                    $(
-                        Self::$variant { .. } => $number,
-                    )*
+                    $(Self::$variant { .. } => $number,)*
                 }
             }
 
             fn format_description(&self) -> String {
                 match self {
-                    $(
-                        Self::$variant { $( $payload_name, )* .. } => format!($description),
-                    )*
+                    $(typedb_error!(@args $variant { $($($arg)*)? }) => format!($description),)*
```

Comment:
I'd prefer to verify here that all non-source fields are used in the formatting, but that would make the macro a lot more complex.